packages/babel-plugin-transform-react-jsx/src/create-plugin.js

```diff
@@ -30,38 +30,59 @@ export default function createPlugin({ name, development }) {
       // TODO (Babel 8): It should throw if this option is used with the automatic runtime
       filter,
 
-      // TODO (Babel 8): Remove `useBuiltIns` & `useSpread`
-      useSpread = false,
-      useBuiltIns = false,
-
       runtime: RUNTIME_DEFAULT = development ? "automatic" : "classic",
 
       importSource: IMPORT_SOURCE_DEFAULT = DEFAULT.importSource,
       pragma: PRAGMA_DEFAULT = DEFAULT.pragma,
       pragmaFrag: PRAGMA_FRAG_DEFAULT = DEFAULT.pragmaFrag,
     } = options;
 
-    // TOOD(Babel 8): If the runtime is 'automatic', we should throw when useSpread/useBuiltIns are set
-    if (RUNTIME_DEFAULT === "classic") {
-      if (typeof useSpread !== "boolean") {
+    if (process.env.BABEL_8_BREAKING) {
+      if ("useSpread" in options) {
         throw new Error(
-          "transform-react-jsx currently only accepts a boolean option for " +
-            "useSpread (defaults to false)",
+          '@babel/plugin-transform-react-jsx: Since Babel 8, an inline object with spread elements is always used, and the "useSpread" option is no longer available. Please remove it from your config.',
```

Comment:
Do we want a similar config message as useBuiltIns here if they want to compile?

Reply:
`@babel/preset-env` already compiles spread by default. We need the message for `useBuiltIns` because by default `...` is not compiled to `Object.assign`.

Reply:
Oh ok great!